relay-cardinality/src/limiter.rs

```diff
@@ -270,25 +291,36 @@ impl<'a, T> CardinalityLimits<'a, T> {
         self.rejections.iter().filter_map(|&i| self.source.get(i))
     }
 
-    /// Consumes the result and returns an iterator over all accepted items.
-    pub fn into_accepted(self) -> Vec<T> {
+    /// Consumes the result and returns [`CardinalityLimitsSplit`] containing all accepted and rejected items.
+    pub fn into_split(self) -> CardinalityLimitsSplit<T> {
         if self.rejections.is_empty() {
-            return self.source;
+            return CardinalityLimitsSplit {
+                accepted: self.source,
+                rejected: Vec::new(),
+            };
         } else if self.source.len() == self.rejections.len() {
-            return Vec::new();
+            return CardinalityLimitsSplit {
+                accepted: Vec::new(),
+                rejected: self.source,
+            };
         }
 
-        self.source
-            .into_iter()
-            .enumerate()
-            .filter_map(|(i, t)| {
+        // TODO: we might want to optimize this method later, by reusing one of the arrays and
+        //  swap removing elements from it.
```

Comment:
```suggestion
        // swap removing elements from it.
```

Reply:
I applied the change but it was done in purpose since it tells the IDE that the following line still belongs to the TODO. No big problem though